src/lib.rs

```diff
@@ -208,18 +221,36 @@ impl<T: Send> ThreadLocal<T> {
     /// Returns the element for the current thread, or creates it if it doesn't
     /// exist. If `create` fails, that error is returned and no element is
     /// added.
+    #[cfg(feature = "nightly")]
+    pub fn get_or_try<F, E>(&self, create: F) -> Result<&T, E>
+    where
+        F: FnOnce() -> Result<T, E>,
+    {
+        if let Some(thread) = unsafe { thread_id::THREAD_HOLDER.as_ref() } {
+            if let Some(inner) = self.get_inner(&thread.0) {
+                return Ok(inner);
+            }
+        }
```

Comment:
Why not just use `thread_id::get()` here? A thread ID is going to need to be allocated anyways.

Reply:
There's no `thread_id::get()` anymore in the nightly version because that used the `thread_local!` macro which introduced unnecessary overhead in the form of at least one branch and other stuff that LLVM wasn't able to optimize out very well. But it is able to do a decent job optimizing with this version where getting/setting the thread local got inlined